vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_unittest.cc

```diff
@@ -0,0 +1,107 @@
+
+namespace ads {
+
+class BatAdsTextEmbeddingHtmlEventsTest : public UnitTestBase {
+ protected:
+  BatAdsTextEmbeddingHtmlEventsTest() = default;
+
+  ~BatAdsTextEmbeddingHtmlEventsTest() override = default;
+};
+
+TEST_F(BatAdsTextEmbeddingHtmlEventsTest, LogEvent) {
+  // Arrange
+  const ml::pipeline::TextEmbeddingInfo text_embedding = BuildTextEmbedding();
+  const std::string embedding_as_string =
+      text_embedding.embedding.GetVectorAsString();
+
+  // Act
+  LogTextEmbeddingHtmlEvent(
+      embedding_as_string, text_embedding.hashed_text_base64,
+      [=](const bool success) {
+        ASSERT_TRUE(success) << "Failed to log text embedding html event";
+        if (!success) {
+          return;
+        }
+
+        GetTextEmbeddingHtmlEventsFromDatabase(
+            [=](const bool success,
+                const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+              ASSERT_TRUE(success)
+                  << "Failed to get text embedding html events";
+              if (!success) {
+                return;
+              }
+
+              // Assert
+              EXPECT_EQ(text_embedding.hashed_text_base64,
+                        text_embedding_html_events[0].hashed_text_base64);
+            });
+      });
+}
+
+TEST_F(BatAdsTextEmbeddingHtmlEventsTest, PurgeEvents) {
+  int n_events_surplus = 4;
+  int n_events =
+      targeting::features::GetTextEmbeddingsHistorySize() + n_events_surplus;
+  int n_events_counter = 0;
+  for (int i = 0; i < n_events; i++) {
+    // Arrange
+    const ml::pipeline::TextEmbeddingInfo text_embedding = BuildTextEmbedding();
+    const std::string embedding_as_string =
+        text_embedding.embedding.GetVectorAsString();
+
+    // Act
+    LogTextEmbeddingHtmlEvent(
+        embedding_as_string, text_embedding.hashed_text_base64,
+        [&](const bool success) {
+          ASSERT_TRUE(success) << "Failed to log text embedding html event";
+          if (!success) {
+            return;
+          }
+
+          n_events_counter++;
+          if (n_events_counter == n_events) {
+            PurgeStaleTextEmbeddingHtmlEvents([](const bool success) {
+              ASSERT_TRUE(success)
+                  << "Failed to purge text embedding html events";
+              if (!success) {
+                return;
+              }
+
+              GetTextEmbeddingHtmlEventsFromDatabase(
+                  [=](const bool success, const TextEmbeddingHtmlEventList&
+                                              text_embedding_html_events) {
+                    ASSERT_TRUE(success)
+                        << "Failed to get text embedding html events";
+                    if (!success) {
+                      return;
+                    }
+
+                    // Assert
+                    const int size_events_default =
+                        targeting::features::GetTextEmbeddingsHistorySize();
+                    const int n_logged_events =
```

Comment:
ideally these should use the names from the functions, i.e. `text_embedding_history_size` and `text_embedding_html_event_count`